### PR TITLE
feat(carSales): create and delete rejections

### PR DESCRIPTION
**Pull request by lkappeler**
refs: https://autoricardo.atlassian.net/browse/CAR-6320

Adds PUT and DELETE actions for carSales rejections so the Dealer can reject draft carSales in the Client.

### CODE REVIEW

**Review by GutierrezGo — COMMENTED** [review]
Not sure if we need the comment and the reason for the reject.

**Comment by lkappeler** [comment]
> Not sure if we need the comment and the reason for the reject.

Yes they where just added, will update the PR

File: src/types/models/carSales.ts [hunk 1]
```diff
@@ -28,3 +28,8 @@ export interface CarSales {
   id: number
   listing: CarSalesListing
 }
+
+export interface CarSaleRejection {
+  comment?: string
+  reason: "car-not-sold" | "wrong-buyer"
+}
```
Comment (GutierrezGo) [inline src/types/models/carSales.ts:35]: I think we need to add `CarSalesRejection` into `CarSales` as `rejection: CarSalesRejection` or maybe I'm missing something there.

Comment (lkappeler) [reply src/types/models/carSales.ts:36]: U right missed that it was missing in the data response as well, added it

File: src/services/reporting/carSales.ts [hunk 2]
```diff
@@ -23,3 +30,71 @@ export const fetchCarSales = async ({
...
+  }
+}
+
+const getRejectionPath = ({
+  dealerId,
+  carSaleId,
+}: {
+  dealerId: number
+  carSaleId: number
+}) => `dealers/${dealerId}/care-sales/${carSaleId}/rejection`
```
Comment (GutierrezGo) [inline src/services/reporting/carSales.ts:55]: I think is `car` intead `care`

**Review by GutierrezGo — APPROVED** [review]

File: src/services/reporting/carSales.ts (hunk 2 above)
Comment (lkappeler) [reply src/services/reporting/carSales.ts]: 🙈  - Good catch!

**Pull request merged**